source/language/subroutines.rst

```diff
@@ -1,6 +1,28 @@
 Subroutines
 ===========
 
+<<<<<<< HEAD
+Subroutines are declared using the statement
+
+    ``def name(parameters) -> output_type { body }``
+
+Subroutines and their named arguments must be named according to the rules for
+identifiers (See :ref:`identifiers`).
+
+Zero or more quantum bits and classical values are passed to the subroutine by reference or name in ``qargs``.
+Classical types are passed by value in ``parameters``.
+
+The subroutines return up to one value of classical type, signified by the
+``return`` keyword. If there is no return type, the empty ``return``
+keyword may be used to immediately exit from the subroutine.
+
+All arguments are declared together with their type, for example ``qubit ancilla``
+would define a quantum bit argument named ``ancilla``. Qubit declarations are not
+allowed within subroutines as those declarations are global.
+
+A subroutine is invoked with the syntax ``name(parameters)`` and may be assigned
+to an ``output`` asneeded via an assignment operator (``=``, ``+=``, etc).
+=======
 Subroutines are declared using the statement ``def name(parameters) -> output_type { body }``.
```

Comment:
I think there's been a bad merge conflict committed here somehow - looks like some of the text that's meant to be changed is duplicated instead as well.

Reply:
Argh, thought I fixed that ... on it right now.